File: src/main/kotlin/common/model/metadata/SuitMetadata.kt
package eelst.ilike.common.model.metadata

interface SuitMetadata {
    val name: String
    val id: String
    val displayName: String
    val abbreviation: String
    val fill: String?
    val fillColors: List<String>
    val clueColors: List<String>
    val createVariants: Boolean
    val pip: String
    val prism: Boolean
    val oneOfEach: Boolean
    val allClueColors: Boolean
    val allClueRanks: Boolean
    val noClueColors: Boolean
    val noClueRanks: Boolean
}

File: src/external/hanablive/dto/metadata/suit_metadata.rs
use serde::Deserialize;

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SuitMetadata {
    pub name: String,
    pub id: String,
    #[serde(default)]
    pub display_name: String,
    #[serde(default)]
    pub abbreviation: String,
    #[serde(default)]
    pub fill: Option<String>,
    #[serde(default)]
    pub fill_colors: Vec<String>,
    #[serde(default)]
    pub clue_colors: Vec<String>,
    #[serde(default)]
    pub create_variants: bool,
    pub pip: String,
    #[serde(default)]
    pub prism: bool,
    #[serde(default)]
    pub one_of_each: bool,
    #[serde(default)]
    pub all_clue_colors: bool,
    #[serde(default)]
    pub all_clue_ranks: bool,
    #[serde(default)]
    pub no_clue_colors: bool,
    #[serde(default)]
    pub no_clue_ranks: bool,
}
